### PR TITLE
Fix #373. add weight for each pattern

### LINKED ISSUE #373: --weight
Let's make it possible to define the weight for certain patterns in the final calculation:

```
$ aibolit recommend --weight=P10:0.5 --weight=P21:1.5
```

### PR DESCRIPTION
**Pull request by lyriccoder**
Fix https://github.com/yegor256/aibolit/issues/373

### CODE REVIEW

**Comment by lyriccoder** [comment]
wait till https://github.com/yegor256/aibolit/pull/381 will be resolved

**Comment by 0crat** [comment]
This pull request [#382](https://github.com/yegor256/aibolit/issues/382) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/83e0587f-a9a3-4e05-a91e-0e07ac589203); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @acheshkov[/z](https://www.0crat.com/u/acheshkov) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: test/recommend/test_recommend_pipeline.py [hunk 1]
```diff
@@ -104,8 +114,23 @@ def __create_input_for_xml(self):
...
     def test_calculate_patterns_and_metrics(self):
+        args = self.__suppress_argparse_mock()
+        file = Path(self.cur_file_dir, 'folder/LottieImageAsset.java')
+        input_params, code_lines_dict, error_string = calculate_patterns_and_metrics(file, args)
+        val = code_lines_dict['P2']
+        self.assertNotEqual(val, 0)
+        val = code_lines_dict['P24']
+        self.assertNotEqual(val, 0)
+
+    def test_calculate_patterns_and_metrics_wih_suppress(self):
```
Comment (aravij) [inline test/recommend/test_recommend_pipeline.py:125]:
```suggestion
    def test_calculate_patterns_and_metrics_with_suppress(self):
```

**Comment by paulodamaso** [comment]
@acheshkov PLease unassign me from this, it's still in a draft stage

**Comment by lyriccoder** [comment]
as discussed with @yegor256  and @acheshkov it's not a issue. We shouldn't set weights manually

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Comment by 0crat** [comment]
Code review was too long (11 days), architects (@acheshkov) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
Pull request [#382](https://github.com/yegor256/aibolit/issues/382) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**